include/nbl/asset/IAssetManager.h

```diff
@@ -320,12 +320,12 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
 
             std::string filePath = _filePath;
             _override->getLoadFilename(filePath, ctx, _hierarchyLevel);
-            system::IFile* file = m_system->createAndOpenFile(filePath.c_str());
 
-            SAssetBundle asset = getAssetInHierarchy_impl<RestoreWholeBundle>(file, _filePath, _params, _hierarchyLevel, _override);
-
-            if (file)
-                file->drop();
+            system::ISystem::future_t<core::smart_refctd_ptr<system::IFile>> future;
+            bool validInput = m_system->createFile(future, _filePath, system::IFile::ECF_READ);
+            assert(validInput);
```

Comment:
what's the boolean and the assertion?

Reply:
`createFile` returns false in case the filename is too long, so i assumed an assertion would be enough

Reply:
how long is too long?

no assertion is not acceptable (it crashes the program, but only in debug mode), you need to fail gracefully.

Reply:
@sadiuk urgent TODO